controller/src/main/java/io/pravega/controller/store/stream/StreamProperty.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+import com.google.common.base.Preconditions;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.io.Serializable;
+
+/**
+ * StreamProperty is an abstraction wrapper over stream properties that can be updated via a workflow.
+ * Namely `configuration` and `truncationRecord`.
+ * StreamProperty is a wrapper over actual property with an `updating` flag that indicates when a property is bring updated.
+ */
+@Data
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class StreamProperty<T> implements Serializable {
+    private final T property;
+    private final boolean updating;
+
+    public static <T> StreamProperty<T> update(final T update) {
+        Preconditions.checkNotNull(update);
+
+        return new StreamProperty<>(update, true);
```

Comment:
This logic is a bit confusing. The state is set to `update` (meaning updated?), but the `updating` flag is set to true. Wouldn't it be simpler to set the property to `updating` rather than `update`?

Reply:
The T update is the updated encapsulated property. 
While the encapsulated property is being updated, the updating flag is set to true. 
This method returns a new StreamProperty object that has the new update for the encapsulated property with updating flag set to true. 

When update is complete, the complete method is called, which sets the updating flag to false.